File: src/0102-binary-tree-level-order-traversal/solution.py
from collections import deque
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        queue = deque()
        res = []

        if root:
            queue.append(root)
        level = 1

        while queue:
            print(f"Current level: {level}")
            level_arr = []
            # empty the queue
            for _ in range(len(queue)):
                curr = queue.popleft()
                level_arr.append(curr.val)
                # add to queue
                if curr.left:
                    queue.append(curr.left)
                if curr.right:
                    queue.append(curr.right)
            level += 1
            res.append(level_arr)
        return res

File: src/0102-binary-tree-level-order-traversal/solution.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function levelOrder(root: TreeNode | null): number[][] {
    const q = new Queue.Queue();
    const result = new Array();
    q.enqueue(root);

    if (!root) {
        return [];
    }

    while (!q.isEmpty()) {
        const val = new Array();
        const queueSize = q.size();
        // this is layer by layer level
        for (let i = 0; i < queueSize; i++) {
            // Go through entire queue, node level
            const curr = q.dequeue();
            val.push(curr.val)
            if (curr.left) {
                q.enqueue(curr.left)
            }
            if (curr.right) {
                q.enqueue(curr.right);
            }
        }
        result.push(val);
    }
    return result;
};
